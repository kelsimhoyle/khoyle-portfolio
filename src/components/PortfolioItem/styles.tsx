import styled from "styled-components";
import { breakpoints } from "../../styles/utils";

export const Cats = styled.div`
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;

    div {
        padding: 2px;
        margin: 0 1px;
    }
`

type PortItem = {
    home: boolean;
}

export const PortItem = styled.div<PortItem>`
    width: 95%;
    margin: ${props => props.home ? "10px" : "20px"} auto;
    display: flex;
    flex-direction: ${props => props.home ? "column" : "row"};
    align-items: center;
    line-height: 1.5;

    ${breakpoints("flex-direction", "", [
        { 800: "column" },
        { 600: "column" },
        { 450: "column" }
      ])};

    p {
        margin: 5px auto;
        line-height: 1.25
    }

    a {
        justify-self: center;
    }

    .image,
    .about {
        margin: 20px;
    }

    .image {
        width: ${({home}) => home ? "95%" : "70%"};
    }
`

export const BtnFlex = styled.div`
    display: flex;
    flex-direction: row;
    width: 100%;
    justify-content: center;
    align-items: stretch;

    > div {
        width: 45%;
    }
`