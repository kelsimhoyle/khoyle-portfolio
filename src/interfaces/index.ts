export interface PortfolioInterface {
    Name: string;
    Slug: string;
    Category: string[];
    Notes: {
        childMarkdownRemark: {
            html: string;
        }
    }
    Image: {
        localFiles: any[]
    };
    Goals?: {
        childMarkdownRemark: {
            html: string;
        }
    }
    Deployed: string;
    Repo: string;
}
