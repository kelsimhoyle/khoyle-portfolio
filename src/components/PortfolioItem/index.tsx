import React from "react";
import { GatsbyImage, getImage } from "gatsby-plugin-image";
import { Link } from "gatsby";
import { PrimaryLink, HightLightedH3, Btn, BtnFlex } from "../../styles";
import { PortItem} from "./styles";

import { PortfolioInterface } from "../../interfaces";

interface PorfolioItemProps {
    data: PortfolioInterface;
    home: boolean;
}

const PortfolioItem = ({ data, home }: PorfolioItemProps) => {
    const { Name, Slug, Category, Notes, Image, Goals, Deployed, Repo } = data;

    return (
        <PortItem home={home} data-testid="item">
            <div className="image-div" data-testid="image-div">
                <GatsbyImage data-testid="image" image={getImage(Image.localFiles[0])} alt={Name} objectFit="contain" />
                <BtnFlex>
                    {Deployed && (
                        <Btn size="small">
                            <a href={Deployed} target="_blank">Visit Website</a>
                        </Btn>
                    )}
                    {Repo && (
                        <Btn size="small">
                            <a href={Repo} target="_blank">GitHub Repo</a>
                        </Btn>
                    )}
                </BtnFlex>
            </div>
            <div className="about" data-testid="about-div">
                <HightLightedH3>{Name}</HightLightedH3>
                <div
                className="goals"
                dangerouslySetInnerHTML={{
                  __html: Notes.childMarkdownRemark.html,
                }}
              />
                <h4>Technologies:</h4>
                <p>{Category.map(cat => ` ${cat} /`)}</p>
               
                <PrimaryLink>
                    <Link to={`/portfolio/${Slug}`}>Learn More</Link>
                </PrimaryLink>
            </div>
        </PortItem>
    );
};

export default PortfolioItem;