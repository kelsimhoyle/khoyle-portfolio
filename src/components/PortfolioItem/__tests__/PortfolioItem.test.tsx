import React from "react";
import renderer from "react-test-renderer"

import PortfolioItem from "../index";
import { PortfolioInterface } from "../../../interfaces";

const data: PortfolioInterface = {
    Name: "James the Dog Walker",
    Slug: "james-the-dog-walker",
    Category: [
        "JavaScript",
        "GraphQL",
        "Gatsby",
        "CMS",
        "API",
        "Custom CMS",
        "Headless",
        "SEO",
        "React",
        "Tailwind CSS",
        "AWS Amplify"
    ],
    Notes: {
        "childMarkdownRemark": {
            "html": "<p>A Southern California dog walking company. Custom headless CMS utilizing Airtable to give client control over content and organize contact requests. Deployed on AWS Amplify. Front end technologies include: React, Gatsby, and Tailwind CSS.</p>"
        }
    },
    Image: {
        localFiles: [
            {
                "childImageSharp": {
                    "gatsbyImageData": {
                        "layout": "constrained",
                        "backgroundColor": "#f8f8f8",
                        "images": {
                            "fallback": {
                                "src": "/static/dca7f4c406b8527e88de0adf9e17a894/db189/NewWebsiteBlueMockupInstagram-Laptop3.png",
                                "srcSet": "/static/dca7f4c406b8527e88de0adf9e17a894/d8461/NewWebsiteBlueMockupInstagram-Laptop3.png 225w,\n/static/dca7f4c406b8527e88de0adf9e17a894/d76bd/NewWebsiteBlueMockupInstagram-Laptop3.png 450w,\n/static/dca7f4c406b8527e88de0adf9e17a894/db189/NewWebsiteBlueMockupInstagram-Laptop3.png 900w",
                                "sizes": "(min-width: 900px) 900px, 100vw"
                            },
                            "sources": [
                                {
                                    "srcSet": "/static/dca7f4c406b8527e88de0adf9e17a894/252a0/NewWebsiteBlueMockupInstagram-Laptop3.webp 225w,\n/static/dca7f4c406b8527e88de0adf9e17a894/2890f/NewWebsiteBlueMockupInstagram-Laptop3.webp 450w,\n/static/dca7f4c406b8527e88de0adf9e17a894/3987a/NewWebsiteBlueMockupInstagram-Laptop3.webp 900w",
                                    "type": "image/webp",
                                    "sizes": "(min-width: 900px) 900px, 100vw"
                                }
                            ]
                        },
                        "width": 900,
                        "height": 900
                    }
                }
            }
        ]
    },
    Goals: {
        "childMarkdownRemark": {
            "html": "<ul>\n<li>Low cost website for potential clients to learn about the company and make contact requests.</li>\n<li>Future plans:\n<ul>\n<li>Custom mobile app to record dog walks for customers to view.\n<ul>\n<li>Mapped walk</li>\n<li>Photo/Video upload</li>\n<li>Notes</li>\n</ul>\n</li>\n</ul>\n</li>\n</ul>"
        }
    },
    Deployed: "https://www.jamesthedogwalker.com/",
    Repo: "https://github.com/kelsimhoyle/james-dog-walker"

};

describe("Portfolio Item", () => {
    it("renders correctly", () => {

        const tree = renderer.create(<PortfolioItem data={data} home={false} />).toJSON()
        expect(tree).toMatchSnapshot()
    })

});
