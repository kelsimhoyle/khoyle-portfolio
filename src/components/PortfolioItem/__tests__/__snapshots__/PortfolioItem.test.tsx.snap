// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio Item renders correctly 1`] = `
<div
  className="sc-kfzBvY iosZMF"
  data-testid="item"
>
  <div
    className="image-div"
    data-testid="image-div"
  >
    <div
      className="gatsby-image-wrapper gatsby-image-wrapper-constrained"
      data-gatsby-image-wrapper=""
      style={
        Object {
          "backgroundColor": undefined,
          "display": "inline-block",
          "overflow": "hidden",
          "position": "relative",
          "verticalAlign": "top",
        }
      }
    >
      <div
        style={
          Object {
            "display": "block",
            "maxWidth": 900,
          }
        }
      >
        <img
          alt=""
          aria-hidden="true"
          role="presentation"
          src="data:image/svg+xml;charset=utf-8,%3Csvg height='900' width='900' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
          style={
            Object {
              "display": "block",
              "maxWidth": "100%",
              "position": "static",
            }
          }
        />
      </div>
      <div
        aria-hidden={true}
        data-placeholder-image=""
        style={
          Object {
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      />
      <picture>
        <source
          data-srcset="/static/dca7f4c406b8527e88de0adf9e17a894/252a0/NewWebsiteBlueMockupInstagram-Laptop3.webp 225w,/static/dca7f4c406b8527e88de0adf9e17a894/2890f/NewWebsiteBlueMockupInstagram-Laptop3.webp 450w,/static/dca7f4c406b8527e88de0adf9e17a894/3987a/NewWebsiteBlueMockupInstagram-Laptop3.webp 900w"
          sizes="(min-width: 900px) 900px, 100vw"
          type="image/webp"
        />
        <img
          alt="James the Dog Walker"
          data-gatsby-image-ssr=""
          data-main-image=""
          data-src="/static/dca7f4c406b8527e88de0adf9e17a894/db189/NewWebsiteBlueMockupInstagram-Laptop3.png"
          data-srcset="/static/dca7f4c406b8527e88de0adf9e17a894/d8461/NewWebsiteBlueMockupInstagram-Laptop3.png 225w,/static/dca7f4c406b8527e88de0adf9e17a894/d76bd/NewWebsiteBlueMockupInstagram-Laptop3.png 450w,/static/dca7f4c406b8527e88de0adf9e17a894/db189/NewWebsiteBlueMockupInstagram-Laptop3.png 900w"
          data-testid="image"
          decoding="async"
          loading="lazy"
          onLoad={[Function]}
          sizes="(min-width: 900px) 900px, 100vw"
          style={
            Object {
              "backgroundColor": undefined,
              "height": "100%",
              "left": 0,
              "objectFit": "contain",
              "objectPosition": undefined,
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transform": "translateZ(0)",
              "transition": "opacity 250ms linear",
              "width": "100%",
              "willChange": "opacity",
            }
          }
        />
      </picture>
      <noscript>
        <picture>
          <source
            sizes="(min-width: 900px) 900px, 100vw"
            srcSet="/static/dca7f4c406b8527e88de0adf9e17a894/252a0/NewWebsiteBlueMockupInstagram-Laptop3.webp 225w,/static/dca7f4c406b8527e88de0adf9e17a894/2890f/NewWebsiteBlueMockupInstagram-Laptop3.webp 450w,/static/dca7f4c406b8527e88de0adf9e17a894/3987a/NewWebsiteBlueMockupInstagram-Laptop3.webp 900w"
            type="image/webp"
          />
          <img
            alt="James the Dog Walker"
            data-gatsby-image-ssr=""
            data-main-image=""
            data-testid="image"
            decoding="async"
            loading="lazy"
            onLoad={[Function]}
            sizes="(min-width: 900px) 900px, 100vw"
            src="/static/dca7f4c406b8527e88de0adf9e17a894/db189/NewWebsiteBlueMockupInstagram-Laptop3.png"
            srcSet="/static/dca7f4c406b8527e88de0adf9e17a894/d8461/NewWebsiteBlueMockupInstagram-Laptop3.png 225w,/static/dca7f4c406b8527e88de0adf9e17a894/d76bd/NewWebsiteBlueMockupInstagram-Laptop3.png 450w,/static/dca7f4c406b8527e88de0adf9e17a894/db189/NewWebsiteBlueMockupInstagram-Laptop3.png 900w"
            style={
              Object {
                "backgroundColor": undefined,
                "height": "100%",
                "left": 0,
                "objectFit": "contain",
                "objectPosition": undefined,
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transform": "translateZ(0)",
                "transition": "opacity 250ms linear",
                "width": "100%",
                "willChange": "opacity",
              }
            }
          />
        </picture>
      </noscript>
      <script
        dangerouslySetInnerHTML={
          Object {
            "__html": "const t=\\"undefined\\"!=typeof HTMLImageElement&&\\"loading\\"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll(\\"img[data-main-image]\\");for(let e of t){e.dataset.src&&(e.setAttribute(\\"src\\",e.dataset.src),e.removeAttribute(\\"data-src\\")),e.dataset.srcset&&(e.setAttribute(\\"srcset\\",e.dataset.srcset),e.removeAttribute(\\"data-srcset\\"));const t=e.parentNode.querySelectorAll(\\"source[data-srcset]\\");for(let e of t)e.setAttribute(\\"srcset\\",e.dataset.srcset),e.removeAttribute(\\"data-srcset\\");e.complete&&(e.style.opacity=1)}}",
          }
        }
        type="module"
      />
    </div>
    <div
      className="sc-hHfuMS jmVEAP"
    >
      <div
        className="sc-kstqJO hvOCEn"
        size="small"
      >
        <a
          href="https://www.jamesthedogwalker.com/"
          target="_blank"
        >
          Visit Website
        </a>
      </div>
      <div
        className="sc-kstqJO hvOCEn"
        size="small"
      >
        <a
          href="https://github.com/kelsimhoyle/james-dog-walker"
          target="_blank"
        >
          GitHub Repo
        </a>
      </div>
    </div>
  </div>
  <div
    className="about"
    data-testid="about-div"
  >
    <h3
      className="sc-hBEYId cAbvZo"
    >
      James the Dog Walker
    </h3>
    <div
      className="goals"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>A Southern California dog walking company. Custom headless CMS utilizing Airtable to give client control over content and organize contact requests. Deployed on AWS Amplify. Front end technologies include: React, Gatsby, and Tailwind CSS.</p>",
        }
      }
    />
    <h4>
      Technologies:
    </h4>
    <p>
       JavaScript /
       GraphQL /
       Gatsby /
       CMS /
       API /
       Custom CMS /
       Headless /
       SEO /
       React /
       Tailwind CSS /
       AWS Amplify /
    </p>
    <span
      className="sc-iqHYmW frONGD"
    >
      <a
        href="/portfolio/james-the-dog-walker"
      >
        Learn More
      </a>
    </span>
  </div>
</div>
`;
